### DISCUSSION TITLE
How to remove dangling images after build a new one?

### DISCUSSION THREAD

**Discussion by thiagoolsilva** [batect/batect#1368] [General]
I've trying to use the feature named dockerfile to build a custom dockerfile image and after ran it I've notice that it creates a untagged images. So, is it possible to reuse the same docker image since any data was changed or delete the untaged image to avoid any trash on my host computer?

**Comment by charleskorn** [comment]
Thanks for the question @thiagoolsilva. This isn't something that Batect currently supports. 

Note that you'll only end up with an untagged image if the definition of the image changes (ie. when the Dockerfile or any files copied into the image change). Rebuilding the same image over and over again should not produce any new images and therefore should not create any garbage.

Docker will also share identical image layers between images wherever possible, so the size reported by `docker images` does not reflect the disk space you'd regain if you were to remove an image. For example, if `image-1` shares a 20 MB layer with `image-2`, and `docker images` shows a size of 23 MB for `image-2`, removing `image-2` while keeping `image-1` will only free the 3 MB consumed by the non-shared layers.

If you'd like an easy way to clean up untagged images, you can run `docker system prune` to remove untagged images, unused networks and stopped containers.

**Reply by thiagoolsilva** [reply]
Thanks @charleskorn
